File: apps/s-2-admin/src/api/credential/Credential.ts
import { JsonValue } from "type-fest";
import { User } from "../user/User";
import { AppModel } from "../appModel/AppModel";
import { DestinationCalendar } from "../destinationCalendar/DestinationCalendar";

export type Credential = {
  id: number;
  typeField: string;
  key: JsonValue;
  user?: User | null;
  appField?: AppModel | null;
  destinationCalendars?: Array<DestinationCalendar>;
};

File: apps/cactus-demo-dotnet/src/APIs/Credential/Dtos/Credential.cs
namespace CactusDemoDotnet.APIs.Dtos;

public class Credential
{
    public int Id { get; set; }

    public string TypeField { get; set; }

    public string Key { get; set; }

    public int? User { get; set; }

    public string? AppField { get; set; }

    public List<int>? DestinationCalendars { get; set; }
}
